types/AbstractBaseType.php

```diff
@@ -242,15 +242,22 @@ public function renderMultiValue($values, \Doku_Renderer $R, $mode) {
      * @return array Tuple with the SQL and parameter array
      */
     public function compare($column, $comp, $value) {
-        if($comp == '~') {
-            $sql = "$column LIKE ?";
-            $opt = array('%' . $value . '%');
-        } else if($comp == '!~') {
-            $sql = "$column NOT LIKE ?";
-            $opt = array('%' . $value . '%');
-        } else {
-            $sql = "$column $comp ?";
-            $opt = array($value);
+        switch ($comp) {
+            case '*~':
+                $sql = "$column LIKE ?";
+                $opt = array('%' . $value . '%');
```

Comment:
This should happen in ConfigParser. When the config parser finds a ``*~`` it simply wraps the value with ``*`` and passes ``~`` as $comp. This way you have one case less in the many Types to handle.